rfcs/datepicker-component.md

```diff
@@ -0,0 +1,130 @@
+# DatePicker Component
+
+Represents control to work and pick date, time value or a range of values
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {DatePicker} from 'baseui/datepicker';
+
+export default () => <DatePicker />;
+```
+
+### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {DatePicker} from 'baseui/datepicker';
+
+export default () => {
+  return <DatePicker
+    initialState={{
+      value: [new Date(2018, 1, 15), new Date(2018, 8, 25)]
+    }}
+    isRange={true}
+    dateFormat={'MM/DD/YY'}
+    minDate={new Date(2000, 1, 1)}
+    maxDate={new Date(2020, 1, 1)}
+    displayMonths={2}
```

Comment:
do we want to document this one too?

Reply:
ok

Reply:
Here are some additional ideas on this option naming: `rangeMode | rangeType | rangeDisplayMode | monthRange: ('single' | 'double') | (1 | 2)`, or can do something like `rangeDoubleMode` to accept a boolean value. Don't have a strong name preference here tho.

Reply:
monthRange so far